sa/storage-authority.go

```diff
@@ -484,54 +525,102 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	return
 }
 
-// AddCertificate stores an issued certificate.
-func (ssa *SQLStorageAuthority) AddCertificate(certDER []byte, regID int64) (digest string, err error) {
-	var parsedCertificate *x509.Certificate
-	parsedCertificate, err = x509.ParseCertificate(certDER)
+// NewPendingCertificate creates a new certificate record with no contents,
+// and returns the resulting empty certificate object.  The incoming certificate
+// object should be empty except for a registration ID.  The returned certificate
```

Comment:
Actually, now I see that we assign the status below, so it's not really an issue.  Conceptually, it seems like it should be allowable for a caller to pass in an empty `core.Certificate` object.